Source/WebCore/platform/graphics/cg/ImageBufferUtilitiesCG.cpp

```diff
@@ -184,6 +186,40 @@ static bool encode(const PixelBuffer& source, const String& mimeType, std::optio
     return encode(image.get(), mimeType, quality, function);
 }
 
+static bool encode(const std::span<const uint8_t>& data, const String& mimeType, std::optional<double> quality, const ScopedLambda<PutBytesCallback>& function)
+{
+    if (data.empty())
+        return false;
+
+    auto destinationUTI = utiFromImageBufferMIMEType(mimeType);
+    if (!destinationUTI)
+        return false;
+
+    auto cfData = adoptCF(CFDataCreateWithBytesNoCopy(nullptr, data.data(), data.size(), kCFAllocatorNull));
+    if (!cfData)
+        return false;
+
+    auto source = adoptCF(CGImageSourceCreateWithData(cfData.get(), nullptr));
+    if (!source)
+        return false;
+
+    CGDataConsumerCallbacks callbacks {
+        [](void* context, const void* buffer, size_t count) -> size_t {
+            auto functor = *static_cast<const ScopedLambda<PutBytesCallback>*>(context);
```

Comment:
what is this doing exactly?

Reply:
See the implementation of `encodeToVector` and the call signature for `CGDataConsumerCreate`. This sets up a callback that will be invoked once transcoding is finished.